apm-sniffer/apm-sdk-plugin/kafka-v1-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/v1/KafkaProducerInterceptor.java

```diff
@@ -47,22 +47,18 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
 
         ProducerRecord record = (ProducerRecord)allArguments[0];
         String topicName = (String)((EnhancedInstance)record).getSkyWalkingDynamicField();
-
+        String key = record.key() == null ? null : (String) record.key();
         AbstractSpan activeSpan = ContextManager.createExitSpan(OPERATE_NAME_PREFIX + topicName + PRODUCER_OPERATE_NAME_SUFFIX, contextCarrier, (String)objInst.getSkyWalkingDynamicField());
 
-        Tags.MQ_BROKER.set(activeSpan, (String)objInst.getSkyWalkingDynamicField());
+        Tags.MQ_BROKER.set(activeSpan, (String) objInst.getSkyWalkingDynamicField());
         Tags.MQ_TOPIC.set(activeSpan, topicName);
+        Tags.MQ_QUEUE.set(activeSpan, key);
         SpanLayer.asMQ(activeSpan);
         activeSpan.setComponent(ComponentsDefine.KAFKA_PRODUCER);
 
-        CarrierItem next = contextCarrier.items();
-        while (next.hasNext()) {
-            next = next.next();
-            record.headers().add(next.getHeadKey(), next.getHeadValue().getBytes());
```

Comment:
AFAIK, kafka 0.11+ supports header. Why remove this?

Reply:
Our online demo is running on kafka.

Reply:
I used kafka-clients of version 0.11.0.3 in my local test, but kafka server use 0.10.0, which will cause error reporting. Do we need to support new clients to connect to old servers?

Reply:
Currently we are using the old version of kafka service in our production cluster, but we also want to use the kafka plugin

Reply:
That is our job. It is Kafka requirement. Nothing more.

If you want, feel free to update document to indicate server version requirement. But it is pointless to make us support less versions tham before. If this change lands, we have broken trace for sure even server is already 0.11+

Reply:
Ok, I get it, I'm going to continue to improve the code

Reply:
> I used kafka-clients of version 0.11.0.3 in my local test, but kafka server use 0.10.0, which will cause error reporting. Do we need to support new clients to connect to old servers?

I think you are mistake. The version has to match with Server and Client. Because Kafka has several MessageFormatVersion. This is the key point.

Reply:
> > I used kafka-clients of version 0.11.0.3 in my local test, but kafka server use 0.10.0, which will cause error reporting. Do we need to support new clients to connect to old servers?
> 
> I think you are mistake. The version has to match with Server and Client. Because Kafka has several MessageFormatVersion. This is the key point.

You can rediscover it by using the kafka service before 0.11 and then using the client after 0.11